### ISSUE TITLE
Question: Is DALI capable for Horovod.torch?

### ISSUE THREAD

**Issue by wantsjean** [NVIDIA/DALI#1357]
I'm using horovod for my distributed training, no matter how the num_shards is set, the size of training set doesnt change. what's wrong?

example of my loader:

    dataloader =FastVideoLoader(batch_size=8, device_id=hvd.rank(), shard_id=hvd.rank(), num_shards=8)

    class FastVideoLoader():
        def __init__(batch_size=8, device_id=hvd.rank(), shard_id=hvd.rank(), num_shards=8):
            self.pipe = VideoPipe(batch_size,shard_id,device_id, num_shards)
            self.pipe.build()
            self.epoch_size = self.pipe.epoch_size("Reader")
            self.dali_iterator = pytorch.DALIGenericIterator(self.pipe,["data","label"],self.epoch_size,auto_reset=True)

        def __len__(self):
            return int(self.epoch_size)

        def __iter__(self):
            return self.dali_iterator.__iter__()

**Comment by awolant** [comment]
Hi, thanks for your question.
I assume `FastVideoLoader` is some kind of custom loader you wrote? Without more complete reproduction is hard to tell what happened there. Could you provide that?

**Comment by wantsjean** [comment]
@awolant I've just updated a simple version of my custom loader.

**Comment by wantsjean** [comment]
@awolant Yes I'm sure.

    output information.
    [1,2]<stdout>:device_id: 2  shard_id: 2  num_shards: 4
    [1,1]<stdout>:device_id: 1  shard_id: 1  num_shards: 4
    [1,3]<stdout>:device_id: 3  shard_id: 3  num_shards: 4
    [1,0]<stdout>:device_id: 0  shard_id: 0  num_shards: 4

**Comment by wantsjean** [comment]
@awolant I've just found in the example that the `epoch_size` should be divided by the `world_size`, thanks for your reply. https://github.com/NVIDIA/DALI/blob/6d4033e218c0d4790d0a0396ffd86b68ad3de938/docs/examples/pytorch/resnet50/main.py#L254

**Comment by JanuszL** [comment]
@wantsjean in the single node it works, but for multinode you should set `device_id=hvd.local_rank(), shard_id=hvd.rank()`.
